packages/wallets/src/signers/shadow-signer/encodeEd25519PublicKey.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * This code was copied from the stellar-sdk-base library.
+ * https://github.com/stellar/js-stellar-base/blob/5bac0b60bcd01794bb0b11fd8b6a45e486c28626/src/strkey.js#L400
+ * This is because the stellar-sdk library is not compatible with React Native.
+ */
+
+import base32 from "base32.js";
+
+function calculateChecksum(payload: Uint8Array): Uint8Array {
```

Comment:
i would double check the implementation from `https://github.com/Crossmint/open-signer/blob/main/shared/cryptography/src/primitives/encoding/strkey.ts` to make sure it's aligned